### PR TITLE
- create Model for Antigen and refactor Feature to accept UniprotKBFeature and AntigenFeature

### PR DESCRIPTION
**Pull request by LeonardoGonzales**
- Create Antigen Model

### CODE REVIEW

**Comment by jluo321** [comment]
AntigenDatabase is not HPA, HPA is evidence type. The AntigenDatabase are ENSG, ENST and ENSP ( ensembl cross reference)

File: core-domain/src/main/java/org/uniprot/core/uniprotkb/feature/UniProtKBFeature.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+public interface UniProtKBFeature extends Feature<UniprotKBFeatureDatabase, UniprotKBFeatureType> {
+
+    FeatureId getFeatureId();
+
+    AlternativeSequence getAlternativeSequence();
+
+    boolean hasFeatureId();
+
+    boolean hasAlternativeSequence();
```
Comment (rizwan-ishtiaq) [inline core-domain/src/main/java/org/uniprot/core/uniprotkb/feature/UniProtKBFeature.java:13]: can't we provide default implementations ?

Comment (LeonardoGonzales) [reply core-domain/src/main/java/org/uniprot/core/uniprotkb/feature/UniProtKBFeature.java:13]: The implementation requires an extra static map that is in the implementation class only.

File: core-domain/src/main/java/org/uniprot/core/antigen/impl/AntigenEntryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    public AntigenEntryBuilder featuresAdd(AntigenFeature feature) {
+        Utils.addOrIgnoreNull(feature, this.features);
+        return this;
+    }
+
+    public AntigenEntryBuilder featuresSet(List<AntigenFeature> features) {
+        this.features = Utils.modifiableList(features);
+        return this;
+    }
+
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply core-domain/src/main/java/org/uniprot/core/antigen/impl/AntigenEntryBuilder.java]: done

**Pull request merged**